### PR TITLE
Remove api from GitLab defaultScope

### PR DESCRIPTION
**Pull request by alexfornuto**
## Summary

Currently, we require a GitLab application with the `api` scope on GitLab (SAAS) to authenticate using it as an IdP. This gives the application full read/write permissions for the entire API, much more than we need.

This change restricts the scope to `email` ~and `openid`~ (already defined). Including `profile` was also discussed, ~but the values therein are already provided by `openid`~ and put back in to match OIDC spec.:

image

I've built and tested this locally, with no problems (other than those caused by my own stupidity :smile: ).

## Related issues

https://github.com/pomerium/docs/issues/25

## Checklist

- [x] reference any related issues
- [ ] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [ ] ready for review

### CODE REVIEW

File: internal/identity/oidc/gitlab/gitlab.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 // Name identifies the GitLab identity provider.
 const Name = "gitlab"
 
-var defaultScopes = []string{oidc.ScopeOpenID, "profile", "email", "api"}
+var defaultScopes = []string{oidc.ScopeOpenID, "email", "openid"}
```
Comment (desimone) [inline internal/identity/oidc/gitlab/gitlab.go:19]:
```suggestion
var defaultScopes = []string{oidc.ScopeOpenID, "email", "profile"}
```
oidc.ScopeOpenID is `openid`

Comment (desimone) [reply internal/identity/oidc/gitlab/gitlab.go:19]: I still feel like `profile` should be included to be in compliance with the openid spec. Unless it doesn't work?

Comment (alexfornuto) [reply internal/identity/oidc/gitlab/gitlab.go]: Dropped the commit.

**Review by calebdoxsey — APPROVED** [review]

File: internal/identity/oidc/gitlab/gitlab.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ import (
 // Name identifies the GitLab identity provider.
 const Name = "gitlab"
 
-var defaultScopes = []string{oidc.ScopeOpenID, "profile", "email", "api"}
+var defaultScopes = []string{oidc.ScopeOpenID, "profile", "email", "openid"}
```
Comment (desimone) [inline internal/identity/oidc/gitlab/gitlab.go]:
```suggestion
var defaultScopes = []string{oidc.ScopeOpenID, "profile", "email"}
```

**Review by desimone — APPROVED** [review]

**Pull request merged**